src/test/libcephfs/ceph_pthread_self.cc

```diff
@@ -0,0 +1,26 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2011 New Dream Network
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#include <pthread.h>
+
+#include <type_traits>
+
+uint64_t ceph_pthread_self() {
+  auto me = pthread_self();
+  static_assert(std::is_convertible_v<decltype(me), uint64_t> ||
+                std::is_pointer_v<decltype(me)>,
+                "we need to use pthread_self() for the owner parameter");
+  return reinterpret_cast<uint64_t>(me);
+}
```

Comment:
pthread_self returns pthread_t on both Linux and BSD, AFAICT. Why is this needed?

Reply:
@jtlayton 
Because the gtest code wants something that resembles a `uint64_t` and  libc shipped with FreeBSD has an opaque `pthread_t`. So conversion needs some helping to get it done.
And this snippet was suggested by @tchaikov, instead of just a raw casting prefix.
And I guess that gcc and glibc are more flexible "to do the right thing"

Reply:
Yuck -- this seems like a bug in gtest, fwiw. In the meantime though this seems like a reasonable workaround. Could you explain this better in the changelog? At some point gtest may stop requiring this, and we may be able to revert this change.